cpp/src/io/orc/stripe_enc.cu

```diff
@@ -687,12 +687,10 @@ __global__ void __launch_bounds__(block_size)
         uint8_t valid           = 0;
         if (row < s->chunk.leaf_column->size()) {
           if (s->chunk.leaf_column->nullable()) {
-            size_type current_valid_offset = row + s->chunk.leaf_column->offset();
-            size_type next_valid_offset =
-              current_valid_offset + min(32, s->chunk.leaf_column->size());
-
-            bitmask_type mask = cudf::detail::get_mask_offset_word(
-              s->chunk.leaf_column->null_mask(), 0, current_valid_offset, next_valid_offset);
+            auto const current_valid_offset = row + s->chunk.leaf_column->offset();
+            auto const last_offset = s->chunk.leaf_column->offset() + s->chunk.leaf_column->size();
```

Comment:
I think this should be limited to 32 bits
```suggestion
            auto const last_offset = min(current_valid_offset + 32, s->chunk.leaf_column->offset() + s->chunk.leaf_column->size());
```

Reply:
This fails because it would always request for a minimum of 32 bits even if we are at the last value and therefore it could request a bitmask-type value beyond the bounds of the validity mask. For example, if we are at `row=8` of a 24-row column (assume offset==0). The code you are suggesting would request bits [8,40) which requires two `bitmask_type` reads from the validity-mask when there is actually only 1 available.

Reply:
We found that since `valid` is `uint8_t` we could further improve this by requesting only 8-bits.